packages/analytics-js/src/services/StoreManager/StoreManager.ts

```diff
@@ -70,30 +71,46 @@ class StoreManager implements IStoreManager {
    */
   initClientDataStore() {
     let storageType: StorageType | '' = '';
+    storageType = state.storage.type.value || '';
 
-    // First try setting the storage to cookie else to localstorage
-    if (getStorageEngine(COOKIE_STORAGE)?.isEnabled) {
-      storageType = COOKIE_STORAGE;
-    } else if (getStorageEngine(LOCAL_STORAGE)?.isEnabled) {
-      storageType = LOCAL_STORAGE;
-    }
-    // TODO: fallback to in-memory storage if not other storage is available
-
-    // TODO: should we fallback to session storage instead so we retain values on page refresh, navigation etc?
-    if (!storageType) {
-      this.logger?.error(STORAGE_UNAVAILABLE_ERROR(STORE_MANAGER));
-      return;
+    switch (storageType) {
+      case COOKIE_STORAGE:
+        if (!getStorageEngine(COOKIE_STORAGE)?.isEnabled) {
```

Comment:
We should also fallback to local storage, similar to the default case. Let's discuss this with the product.